File: src/main/java/org/pl/lexer/tokenizer/CommaTokenizer.java
package org.pl.lexer.tokenizer;

import org.pl.lexer.ILexer;
import org.pl.lexer.exception.TokenizerException;
import org.pl.lexer.token.CommaToken;

/**
 * Tokenizes the symbol ','.
 */
public class CommaTokenizer implements ITokenizer {
    @Override
    public boolean matches(Character c) {
        return c == ',';
    }

    @Override
    public CommaToken tokenize(ILexer lexer) throws TokenizerException {
        return new CommaToken();
    }
}

File: src/main/kotlin/org/nyxlang/lexer/tokenizer/CommaTokenizer.kt
package org.nyxlang.lexer.tokenizer

import org.nyxlang.lexer.ILexer
import org.nyxlang.lexer.token.CommaToken

/**
 * Tokenizes the symbol ','.
 */
class CommaTokenizer : ITokenizer {
    override fun matches(c: Char) = c == ','
    override fun tokenize(lexer: ILexer) = CommaToken()
}